src/libraries/System.Security.Cryptography.Xml/src/System/Security/Cryptography/Xml/XmlDecryptionTransform.cs

```diff
@@ -150,8 +150,8 @@ private void LoadStreamInput(Stream stream)
         {
             XmlDocument document = new XmlDocument();
             document.PreserveWhitespace = true;
-            XmlResolver resolver = (ResolverSet ? _xmlResolver : new XmlSecureResolver(new XmlUrlResolver(), BaseURI));
-            XmlReader xmlReader = Utils.PreProcessStreamInput(stream, resolver, BaseURI);
+            XmlResolver resolver = ((ResolverSet ? _xmlResolver : new XmlSecureResolver(new XmlUrlResolver(), BaseURI))!);
```

Comment:
The extra parens don't seem necessary... the `!` could just be added outside the existing ones... or maybe just after the first clause?

```suggestion
            XmlResolver resolver = (ResolverSet ? _xmlResolver! : new XmlSecureResolver(new XmlUrlResolver(), BaseURI));
```

Reply:
I went a step further and borrowed a technique I discovered earlier in the PR: I decorated `ResolverSet` with `[MemberNotNullWhen(true, nameof(_xmlResolver))]`
This meant that a lot of the extra parens and bangs melted away - which was very satisfying!